flink-clients/src/main/java/org/apache/flink/client/FlinkYarnSessionCli.java

```diff
@@ -417,13 +417,9 @@ public int run(String[] args) {
 			String jobManagerAddress = yarnCluster.getJobManagerAddress().getAddress().getHostAddress() + ":" + yarnCluster.getJobManagerAddress().getPort();
 			System.out.println("Flink JobManager is now running on " + jobManagerAddress);
 			System.out.println("JobManager Web Interface: " + yarnCluster.getWebInterfaceURL());
-			// file that we write into the conf/ dir containing the jobManager address and the dop.
-
-			String defaultPropertiesFileLocation = System.getProperty("java.io.tmpdir");
-			String currentUser = System.getProperty("user.name");
-			String propertiesFileLocation = yarnCluster.getFlinkConfiguration().getString(ConfigConstants.YARN_PROPERTIES_FILE_LOCATION, defaultPropertiesFileLocation);
 
-			File yarnPropertiesFile = new File(propertiesFileLocation + File.separator + CliFrontend.YARN_PROPERTIES_FILE + currentUser);
+			// file that we write into the conf/ dir containing the jobManager address and the dop.
+			File yarnPropertiesFile = new File(getYarnPropertiesLocation(yarnCluster.getFlinkConfiguration()));
```

Comment:
The yarnPropertiesFile path construction is also done in `CliFrontend:173`. This should also be changed to use the new `getYarnPropertiesLocation` method. Maybe moving this method to `CliFrontend` makes sense.